Library/Homebrew/cmd/list.rb

```diff
@@ -122,7 +122,13 @@ def run
 
           raise UsageError, "Cannot use #{flags.join(", ")} with formula arguments." unless args.no_named?
 
-          Formula.installed.sort.each do |formula|
+          formulae = if args.t?
+            Formula.installed.sort_by { |formula| test("M", formula.prefix.realpath) }.reverse!
```

Comment:
```suggestion
            Formula.installed.sort_by { |formula| test("M", formula.rack) }.reverse!
```

Probably closer to what `ls` is doing.